### ISSUE TITLE
Error in custom fetch function not caught

### ISSUE THREAD

**Issue by wereHamster** [FormidableLabs/urql#517]
**Describe the bug**

I am providing a custom `fetch` function when creating the client. This function may throw an custom error. When it does so the query inside the app is never resolved, it stays loading forever.

**Steps to reproduce**

```
const client = createClient({
  fetch: async () => { throw "FAIL" }
})

const [result] = useQuery(…)
```

Example on codesandbox here: https://codesandbox.io/s/purple-meadow-ynux5.

Your codesandbox template uses an older version of urql (1.2.0) with that old version it works. After upgrading to 1.8.2 the bug is reproducible.

**Expected behavior**

The result is first `{ loading: true, error: undefined }` and then turns to `{ loading: false, error: CombinedError(…) }`.

**Actual behavior**

The result stays loading forever.

**Comment by JoviDeCroock** [comment]
Hey @wereHamster 

Thanks for notifying us about the codesandbox template being outdated, we've updated this! Thanks for the bug reproduction as well. We'll certainly look into this

**Comment by wereHamster** [comment]
Bug not reproducible in 1.7.0. It was introduced in 1.8.0.

**Comment by JoviDeCroock** [comment]
Thanks for narrowing it down, this might be related to our `wonka@4` upgrade then

**Comment by wereHamster** [comment]
I was able to trace the correct response context (with a valid CombinedError) through the cache/dedup exchange, but the it became too difficult to trace the code execution (stepping through the compiled wonka code is no fun).

**Comment by JoviDeCroock** [comment]
Hey @wereHamster 

This seems fixed in 1.9.0, can you confirm this helps your case?

**Comment by wereHamster** [comment]
Yep, can confirm that it works in 1.9.0. Thanks.
